programs/uxd/src/instructions/lsd/borrow_from_lsd_depository.rs

```diff
@@ -0,0 +1,144 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+
+use crate::error::UxdError;
+use crate::state::LsdDepository;
+use crate::state::LsdPosition;
+use crate::utils::validate_collateral_amount;
+use crate::utils::validate_loan_to_value_bps;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::CONTROLLER_NAMESPACE;
+use crate::LSD_DEPOSITORY_NAMESPACE;
+use crate::LSD_POSITION_SPACE;
+use crate::LSD_PROFITS_TOKEN_ACCOUNT_NAMESPACE;
+
+#[derive(Accounts)]
+pub struct BorrowFromLsdDepository<'info> {
+    /// #1
+    pub user: Signer<'info>,
+
+    /// #2
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        has_one = redeemable_mint @UxdError::InvalidRedeemableMint
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #4
+    #[account(
+        mut,
+        seeds = [LSD_DEPOSITORY_NAMESPACE, collateral_mint.key().as_ref()],
+        bump = depository.load()?.bump,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint
+    )]
+    pub depository: AccountLoader<'info, LsdDepository>,
+
+    /// #5
+    #[account(
+        mut,
+        seeds = [LSD_PROFITS_TOKEN_ACCOUNT_NAMESPACE, depository.key().as_ref()],
+        token::authority = depository,
+        token::mint = liquidation_mint,
+        bump = depository.load()?.profits_token_account_bump,
+    )]
+    pub profits_token_account: Account<'info, TokenAccount>,
+
+    /// #6
+    #[account(
+        mut,
+        constraint = user_collateral.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_collateral.mint == collateral_mint.key() @UxdError::InvalidCollateralMint,
+    )]
+    pub user_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #
+    #[account(
+        mut,
+        constraint = user_redeemable.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_redeemable.mint == redeemable_mint.key() @UxdError::InvalidRedeemableMint,
+    )]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+
+    /// #
+    /// Tracks the user overall position in the LSD Depository (one per user, per LSD Depository).
+    #[account(
+        init_if_needed,
+        seeds = [depository.key().as_ref(), user.key().as_ref()],
+        space = LSD_POSITION_SPACE,       
+        payer = payer,
+        bump,
+    )]
+    pub lsd_position: AccountLoader<'info, LsdPosition>,
+
+    /// #
+    /// PDA holding the deposited collateral on behalf of the lsd_position
+    #[account(
+        init_if_needed,
+        seeds = [lsd_position.key().as_ref()],
+        token::authority = lsd_position,
+        token::mint = collateral_mint,
+        payer = payer,
+        bump,
+    )]
+    pub lsd_position_collateral_token_account: Box<Account<'info, TokenAccount>>,
```

Comment:
micro-nit: in other places in the code, we would just call this:

`lsd_position_collateral`

Reply:
renamed